includes/Jobs/RequestWikiRemoteAIJob.php

```diff
@@ -0,0 +1,395 @@
+
+	private Config $config;
+	private WikiRequestManager $wikiRequestManager;
+	private HttpRequestFactory $httpRequestFactory;
+	private LoggerInterface $logger;
+	private string $baseApiUrl;
+	private string $apiKey;
+
+	private int $id;
+	private string $reason;
+	private string $sitename;
+	private string $subdomain;
+
+	public function __construct(
+		array $params,
+		ConfigFactory $configFactory,
+		WikiRequestManager $wikiRequestManager,
+		HttpRequestFactory $httpRequestFactory
+	) {
+		parent::__construct( self::JOB_NAME, $params );
+
+		$this->config = $configFactory->makeConfig( 'CreateWiki' );
+		$this->wikiRequestManager = $wikiRequestManager;
+		$this->httpRequestFactory = $httpRequestFactory;
+		$this->logger = LoggerFactory::getInstance( 'CreateWiki' );
+
+		$this->baseApiUrl = 'https://api.openai.com/v1';
+		$this->apiKey = $this->config->get( ConfigNames::OpenAIConfig )['apikey'] ?? '';
+
+		$this->id = $params['id'];
+		$this->reason = $params['reason'];
+		$this->sitename = $params['sitename'];
+		$this->subdomain = $params['subdomain'];
+	}
+
+	public function run(): bool {
+		$services = MediaWikiServices::getInstance();
+
+		if ( !$this->config->get( ConfigNames::OpenAIConfig )['apikey'] ) {
+			$this->logger->debug( "OpenAI API key is missing! AI job cannot start." );
+			throw new ConfigException( 'OpenAI API key is missing! Cannot query API without it!' );
+		} elseif ( !$this->config->get( ConfigNames::OpenAIConfig )['assistantid'] ) {
+			$this->logger->debug( "OpenAI Assistant ID is missing! AI job cannot start." );
+			throw new ConfigException( 'OpenAI Assistant ID is missing! Cannot run AI model without an assistant!' );
+		}
+
+		$this->wikiRequestManager->loadFromID( $this->id );
+		$this->logger->debug( "Loaded request {$this->id} for AI approval." );
+
+		if ( !$this->canAutoApprove() ) {
+			$this->logger->debug( "Wiki request {$this->id} was not auto-evaluated due to denylist." );
+			return true;
+		}
+
+		// Initiate OpenAI query for decision
+		$this->logger->info( "Querying OpenAI for decision on wiki request {$this->id}..." );
+		$apiResponse = $this->queryOpenAI( $this->sitename, $this->subdomain, $this->reason );
+
+		if ( !$apiResponse ) {
+			return true;
+		}
+
+		// Extract response details with default fallbacks
+		$outcome = $apiResponse['recommendation']['outcome'] ?? 'reject';
+		$comment = $apiResponse['recommendation']['public_comment'] ?? 'No comment provided. Please check logs.';
+
+		$this->logger->info( "AI decision: {$outcome} for request {$this->id}. Comment: {$comment}" );
+
+		if ( $this->config->get( ConfigNames::OpenAIConfig )['dryrun'] ) {
+			return $this->handleDryRun( $outcome, $comment );
+		} else {
+			return $this->handleLiveRun( $outcome, $comment );
+		}
+	}
+
+	private function handleDryRun( string $outcome, string $comment ): bool {
+		$services = MediaWikiServices::getInstance();
+		$outcomeMessage = Message::newFromKey( "requestwikiqueue-$outcome" )->text();
```

Comment:
add a $this->messageLocalizer in constructor get from RequestContext::getMain() but set $this->messageLocalizer to it and type hint as MessageLocalizer in property. Also 'requestwikiqueue' . $outcome. Do this for all other similar also.